docs/design/flow-sla/README.md

```diff
@@ -0,0 +1,96 @@
+# SLA for flows in Kilda
+
+## Problem
+
+Since the first release there were only one SLA for Kilda flows - bandwidth. But now there are more sla's that are taken
+into account for the flows. E.g. latency, latency_tier2. The purpose of this doc is to rethink and redesign how these 
+SLAs are visible outside the system and define strategy for adding new ones.
+
+## Current State
+
+Here is the list of flow fields that are taken into account by path computer:
+- maximum bandwidth
+- max_latency
+- max_latency_tier2 
+- diverse_flow_id
+- affinity_flow_id
+- allocate_protected_path
+- ignore_bandwidth
+- strict_bandwidth
+
+Planned to add:
+- max_bandwidth_tier2 
+
+All these fields are part of flow entity, which is not helpful from operational perspective and also isn't practical
+from visibility perspective.
+
+## Proposal
+
+Introduce new Entity in API and Core layer which will group all SLA related fields for the object. Add new API, which
+increase visibility of the flow SLA's status
+
+### Entity:
+
+`FlowSla` with the following list of fields:
+- bandwidth_tier1
+- bandwidth_tier2
+- latency_tier1
+- latency_tier2
+- affinity_flow_id
+- diversity_flow_id
+- protected_path
+- strict_bandwidth
+
+
+## API
+
+added new field to flow entity in crud operations
+
+### POST / PUT / PATCH  
+```
+{
+  "flow_id": "string",
+  ...
+  "sla": {
+    "bandwidth_tier1": 0,
+    "bandwidth_tier2": 0,
+    "latency_tier1": 0, 
+    "latency_tier2": 0,
+    "affinity_flow_id": "string",
+    "diversity_flow_id": "string",
+    "protected_path": true,
+    "strict_bandwidth": true,
+  }
```

Comment:
If we are to rethink this completely, then I would have waived the tier1/tier2 approaches and introduced  'fallback' SLA objects. If prio:1 SLA has failed, try to apply next priority SLAs and apply Degraded status to the flow (maybe even introduce the "degradationLevel" value that will be scored according to the SLA priority that was used to build the flow). If none can be applied - flow creation is denied completely (or flow reroute makes it Down)
```
  "slas": [
         {
             "bandwidth": 100,
             "latency": 100, 
             "affinity_flow_id": "string",
             "diversity_flow_id": "string",
             "protected_path": true,
             "ignore_bandwidth": false
             "priority": 1
         },
          {
             "bandwidth": 50,
             "latency": 200, 
             "affinity_flow_id": "string",
             "protected_path": false,
             "ignore_bandwidth": false,
             "priority": 2
         },
          {
             "bandwidth": 50,
             "latency": 300, 
             "protected_path": false,
             "ignore_bandwidth": true,
             "priority": 3
         }
]
```

Reply:
This may also be used to simplify all our 'retry' strategies. (retry with ignore bandwidth, retry with latency_tier2 etc.). System will just cycle through available SLAs